packages/browser-wallet/src/background/recovery.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    createIdentityRecoveryRequest,
+    CryptographicParameters,
+    IdentityObjectV1,
+    IdentityRecoveryRequestInput,
+    Network as NetworkString,
+    Versioned,
+} from '@concordium/web-sdk';
+import { ExtensionMessageHandler } from '@concordium/browser-wallet-message-hub';
+import { IdentityProviderIdentityStatus, IdentityTokenContainer, sleep } from 'wallet-common-helpers';
+import { Identity, CreationStatus, IdentityProvider } from '@shared/storage/types';
+import { BackgroundResponseStatus } from '@shared/utils/types';
+import { addIdentity } from './update';
+
+// TODO: increase max empty
+// How many empty identityIndices are allowed before stopping
+const maxEmpty = 3;
+// Milliseconds to wait, if retrievelUrl is still pending
+const sleepInterval = 5000;
+
+export type Payload = {
+    providers: IdentityProvider[];
+    globalContext: CryptographicParameters;
+    seedAsHex: string;
+    net: NetworkString;
+};
+
+/**
+ * Polls the provided location until a valid identity object is available, or that an error is returned.
+ * If an error occurs or is returned from the url, this return undefined
```

Comment:
This is not currently true. You need to guard against all errors with a `try-catch` for it to work like that. I guess in the context of recovery it would make sense to fail the whole recovery if they return an 404 or whatever because they are unavailable.